d2/src/test/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancerStateTest.java

```diff
@@ -1469,7 +1469,7 @@ public void testSSLDisabledWithHttpsInstances() throws URISyntaxException
     assertNull(client, "shouldn't pick an https uri");
 
     _state.refreshClients(serviceProperties);
-
+    assertEquals(_state.getUriProperties("cluster-1").getVersion(), _state.getVersion());
```

Comment:
Since it's a race condition, is it better to run the test multiple times?

Reply:
It's very hard to reproduce the issue within unit tests. But I manually reproduced the issue by adding some time delay and verified the fix.